parsel/utils.py

```diff
@@ -57,15 +57,20 @@ def _is_listlike(x: Any) -> bool:
 
 
 def extract_regex(
-    regex: Union[str, Pattern[str]], text: str, replace_entities: bool = True
+    regex: Union[str, Pattern[str]],
+    text: str,
+    replace_entities: bool = True,
+    flags: int = 0,
 ) -> List[str]:
     """Extract a list of unicode strings from the given text/encoding using the following policies:
+    * if the regex is a string it will be compiled using the provided flags
     * if the regex contains a named group called "extract" that will be returned
     * if the regex contains multiple numbered groups, all those will be returned (flattened)
     * if the regex doesn't contain any group the entire regex matching is returned
     """
     if isinstance(regex, str):
-        regex = re.compile(regex, re.UNICODE)
+        flags |= re.UNICODE
+        regex = re.compile(regex, flags)
```

Comment:
I keep the `re.UNICODE` to respect the old behavior, especially because in some docstrings you can find:
> Apply the given regex and return the first unicode string which

However, this also means that it won't be possible to override this, and the `re.UNICODE` will be always applied. I don't think this is a big issue and can be changed in the future, for example when deprecating Python 2.7.

Reply:
I've seen that Python 2.7 has been already deprecated...

Reply:
The Unicode flag also applies to Python 3, though, is not about the Python 2 strings but about pattern behavior.

On the other hand, I see that we actually compile strings into patterns here. It’s out of the scope of this change, so feel free to ignore, but I wonder if, instead of compiling the expressions, we could pass them (and `flags`) to the corresponding functions. That may even allow flags to work when passed along a compiled regular expression.

Reply:
> The Unicode flag also applies to Python 3, though, is not about the Python 2 strings but about pattern behavior.

Sorry, I think I wasn't clear enough and mixed things. What I meant was that we could do something like:
```python
regex = re.compile(regex, flags or re.UNICODE)
```
or doing what I did (always applying `re.UNICODE`).  The first approach allows us to override the `re.UNICODE`, but if we were applying `re.UNICODE` always, it could be confusing to don't apply it when using other flags. The reference I made to Python 2.7 was that we could change this in the future but it would be a breaking change, however, it could come along with the Python 2.7 deprecation that would require a new major version. I didn't know it had been already deprecated before; after I saw that, I considered that maybe we need to think better about this decision.

> On the other hand, I see that we actually compile strings into patterns here. It’s out of the scope of this change, so feel free to ignore, but I wonder if, instead of compiling the expressions, we could pass them (and flags) to the corresponding functions. That may even allow flags to work when passed along a compiled regular expression.

It seems a good idea, but if it's not required I would love to keep this PR as-is (shorter). We can open a new issue if you want :)